qualtran/bloqs/basic_gates/hadamard.py

```diff
@@ -0,0 +1,71 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property, reduce
+from typing import Dict, Tuple
+
+import cirq
+import numpy as np
+import quimb.tensor as qtn
+from attrs import frozen
+
+from qualtran import Bloq, Signature, SoquetT
+from qualtran.cirq_interop import CirqQuregT
+
+_HADAMARD = np.array([[1.0, 1.0], [1.0, -1.0]]) / (2**0.5)
+
+
+@frozen
+class Hadamard(Bloq):
+    """The Hadamard gate.
+
+    Args:
+        bitsize: The number qubits to add a Hadamard gate to.
```

Comment:
be explicit that multi-qubit H is the same as H on each qubit.

---

I'm torn about whether we should do it this way or introduce an `OnEach` meta-bloq. The reason why having bitsize be acceptable here is because "hadamard matrix" in math is a family of different sized matrices (although they don't l2-normalize theirs).